### PR TITLE
Added This Code

### PR DESCRIPTION
**Pull request by Sinchan26**
### **Describe your change:**

- [ ] Add an algorithm?
- [ ] Fix a bug or typo in an existing algorithm?
- [ ] Documentation change?
- [x] Add a code?
### **Checklist:**

- [ x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Java/blob/master/CONTRIBUTING.md).
- [ x] This pull request is all my own work -- I have not plagiarized.
- [x ] I know that pull requests will not be merged if they fail the automated tests.
- [ x] This PR only changes one algorithm file. To ease review, please open separate PRs for separate algorithms.
- [x ] All new Java files are placed inside an existing directory.
- [x ] All filenames are in all uppercase characters with no spaces or dashes.
- [x ] All functions and variable names follow Java naming conventions.
- [x ] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
- [x ] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: src/main/java/com/thealgorithms/dynamicprogramming/ClimbingStairs.java [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+/*
+ * Output:-
+13
+34
+233
+10946
+377
+
+Time-Complexity:-
+ */
```
Comment (Sinchan26) [inline src/main/java/com/thealgorithms/dynamicprogramming/ClimbingStairs.java:62]: O(n)

**Review by siriak — CHANGES_REQUESTED** [review]
Could you please add some JUnit tests and remove `main()`? (see https://github.com/TheAlgorithms/Java/tree/master/src/test/java/com/thealgorithms)

File: src/main/java/com/thealgorithms/dynamicprogramming/ClimbingStairs.java [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+        }
+        return dp[n];
+    }
+    public static void main(String[] args) {
+        System.out.println(climbStairs(6));
+        System.out.println(climbStairs(8));
+        System.out.println(climbStairs(12));
+        System.out.println(climbStairs(20));
+        System.out.println(climbStairs(13));
+    }
```
Comment (siriak) [inline src/main/java/com/thealgorithms/dynamicprogramming/ClimbingStairs.java:51]: Please remove this method

File: src/test/java/com/thealgorithms/dynamicprogramming/ClimbingStairsTest.java [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    public static int climbStairs(int n) {
+        int[] dp=new int[n+1];
+        dp[0]=1;
+        dp[1]=1;
+       
+        for(int i=2;i<=n;i++){
+            dp[i]=dp[i-1]+dp[i-2];
+        }
+        return dp[n];
+    }
```
Comment (siriak) [inline src/test/java/com/thealgorithms/dynamicprogramming/ClimbingStairsTest.java:18]: Please reference the original method instead of copying it

**Review by siriak — CHANGES_REQUESTED** [review]

**Pull request closed without merging**